File: Data Structures & Algorithms/majority-element/submission-3.java
class Solution {
    public int majorityElement(int[] nums) {
        int majEle = 0;
        int count = 0;
        for(int i=0;i<nums.length;i++){
            if(count == 0){
                majEle = nums[i];
            }
            if(majEle != nums[i]){
                count--;
            }else {
                count++;
            }
        }
        return majEle;
    }
}

File: Data Structures & Algorithms/sort-colors/submission-3.js
class Solution {
    sortColors(nums) {
        let low = 0;
        let mid = 0;
        let high = nums.length - 1;

        while (mid <= high) {
            if (nums[mid] === 0) {
                this.swap(nums, low, mid);
                low++;
                mid++;
            } else if (nums[mid] === 1) {
                mid++;
            } else {
                this.swap(nums, mid, high);
                high--;
            }
        }
    }

    swap(nums, a, b) {
        [nums[a], nums[b]] = [nums[b], nums[a]];
    }
}